---
title: "How-to-use"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{How-to-use}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# Installation
```{r setup, eval=FALSE}
devtools::install_github("nimahemmat/CNASigDetect")
```

After installation, run the library
```{r loading}
library(CNASigDetect)
```

# Proccesing inputs

In order to test the functions, a table of segments from TCGA-BRCA dataset is available as `TCGA_BRCA_Segments`.
```{r}
head(TCGA_BRCA_Segments)
```


For an optimal execution of package, it is recommended to prepare input file as below:
```{r}
input_segments <- MakeSegInput(Segments     = TCGA_BRCA_Segments,
                               SampleID_col = "SampleID", 
                               Chr_col      = "Chromosome", 
                               Start_col    = "Start", 
                               End_col      = "End", 
                               MajorCN_col  = "Major_Copy_Number", 
                               MinorCN_col  = "Minor_Copy_Number")
head(input_segments)
```

In this function, you can easily create an input file based on your data frame of allele-specific segments.

`Segments` is the name of your segment variable which should be a data frame.

Within your data frame, `SampleID_col` is the name of corresponding column for the sample IDs, `Chr_col` is that for chromosome name column, `Start_col` and `End_col` are the start and end column of segments, `MajorCN_col` and `MinorCN_col` are the name of columns for major and minor copy numbers.

# Copy number signatures

`ExtractCnaSignature()` function is used to classify segments based on their copy numbers and length, and generate a list of results contains the features for each constructed CNA signature and contribution of each signature across the cohort.

To run this function, follow:
```{r}
# Should load NMF before this function
library(NMF)
result_signatures <- ExtractCnaSignature(Input.Segments = input_segments, 
                                         Focal_th       = 3,
                                         Broad_th       = 10,
                                         Art.len        = 0.05,
                                         k_range        = 2:10,
                                         n_run          = 100)
```

The arguments needed for this function are `Input.Segments` which is the data frame created by `MakeSegInput()`, `Focal_th` which is the threshold of length in Mb for focal alterations, `Broad_th` which is the threshold of length in Mb for broad alterations, and `Art.len` which is the threshold for artifact segments which are mainly due to hypersegmentation. Segments which are between focal and broad thresholds are called medium segments. Other arguments are `k_range` and `n_run` which are for NMF test.

```{r}
# Features of constructed signatures.
signatures_features <- result_signatures[["Signature_Annotation"]]
head(signatures_features)
```

```{r}
# Contribution of each signature across the cohort
signatures_contribution <- result_signatures[["Sample_Weights"]]
head(signatures_contribution)
```

## Plot signatures

To visualise signatures' features, you can pass `ExtractCnaSignature()` results into `PlotSignatures()` as below:
```{r}
signatures_plots <- PlotSignatures(Sig.Results = result_signatures)
```

Result is a list of bar plots for each signature:
```{r}
signatures_plots[[1]]
```

# Cytoband level alteration
`CytobandLevelAlteration()` function could accurately detect cytoband level alteration using input file generated by `MakeSegInput()`. This function calculates the affected length of each cytoband and estimates the length-weighted alterations per cytoband in each sample. Currently, this function support both hg38 and hg19 reference genome.

```{r}
cytoband_level <- CytobandLevelAlteration(Input.Segments = input_segments,
                                          Genome         = "hg19",
                                          Art.len        = 0.05)
head(cytoband_level)
```

The arguments of function are `Input.Segments` which is the data frame generated by `MakeSegInput()`, `Genome` which can be either `hg19` or `hg38`, and `Art.len` which is the threshold for artifact segments.

The output file has the below column:

| Column            | Description                                                                         |
|-------------------|-------------------------------------------------------------------------------------|
| SampleID          | Unique identifier of each sample                                                    |
| band_chr          | Chromosome name                                                                     |
| band_start        | Start point of cytoband                                                             |
| band_end          | End point of cytoband                                                               |
| band              | Name of cytoband                                                                    |
| band_span_mb      | Length of cytoband in Mb                                                            |
| covered_mb        | Length of cytoband covered by corresponding segment in Mb                           |
| frac_band_covered | Length of cytoband covered by corresponding segment in Mb                           |
| tcn_lenw          | Length-weighted total copy number of cytoband                                       |
| lcn_lenw          | Length-weighted minor copy number of cytoband                                       |
| frac_del_lenw     | Length-weighted fraction of cytoband which is deleted                               |
| frac_loh_lenw     | Length-weighted fraction of cytoband which harboures LOH                            |
| frac_gain_lenw    | Length-weighted fraction of cytoband which has 2 < copy number < 7                  |
| frac_amp_mod_lenw | Length-weighted fraction of cytoband which has 6 < copy number < 10                 |
| frac_amp_hi_lenw  | Length-weighted fraction of cytoband which has copy number more than or equal to 10 |

## Genome wide alterations plot

In order to visualise genome wide alterations for Gain/Loss, Amplification/Deletion, and rate of LOH across the cohort `PlotGenomeWideAlteration()` can be utilised using cytoband level data frame created by `CytobandLevelAlteration()`.

```{r}
genome_wide_plots <- PlotGenomeWideAlteration(Cytoband.level.data = cytoband_level, 
                                              Genome              = "hg19", 
                                              Coverage            = 0.8)
```

This function generates a list of Gain/Loss, Amp/Del, LOH rate plots, and a data frame which contains the frequnecy of alterations per cytoband.

```{r}
genome_wide_plots[["GainLoss_plot"]]
```

```{r}
genome_wide_plots[["AmpDel_plot"]]
```

```{r}
genome_wide_plots[["LOH_plot"]]
```

```{r}
head(genome_wide_plots[["Cytobands_frq"]])
```
