<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>How-to-use • CNASigDetect</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="How-to-use">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNASigDetect</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/How-to-use.html">How-to-use</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>How-to-use</h1>
            
      

      <div class="hidden name"><code>How-to-use.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"nimahemmat/CNASigDetect"</span><span class="op">)</span></span></code></pre></div>
<p>After installation, run the library</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CNASigDetect</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="proccesing-inputs">Proccesing inputs<a class="anchor" aria-label="anchor" href="#proccesing-inputs"></a>
</h2>
<p>In order to test the functions, a table of segments from TCGA-BRCA
dataset is available as <code>TCGA_BRCA_Segments</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">TCGA_BRCA_Segments</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SampleID Chromosome   Start     End Major_Copy_Number</span></span>
<span><span class="co">#&gt; 1 TCGA-D8-A27H-01A       chr1   13116  732994                 3</span></span>
<span><span class="co">#&gt; 2 TCGA-D8-A27H-01A       chr1  734210 1942014                 2</span></span>
<span><span class="co">#&gt; 3 TCGA-D8-A27H-01A       chr1 1943142 1987886                 2</span></span>
<span><span class="co">#&gt; 4 TCGA-D8-A27H-01A       chr1 1989523 6821628                 2</span></span>
<span><span class="co">#&gt; 5 TCGA-D8-A27H-01A       chr1 6822926 6862432                 1</span></span>
<span><span class="co">#&gt; 6 TCGA-D8-A27H-01A       chr1 6863666 6874235                 2</span></span>
<span><span class="co">#&gt;   Minor_Copy_Number</span></span>
<span><span class="co">#&gt; 1                 0</span></span>
<span><span class="co">#&gt; 2                 2</span></span>
<span><span class="co">#&gt; 3                 1</span></span>
<span><span class="co">#&gt; 4                 2</span></span>
<span><span class="co">#&gt; 5                 1</span></span>
<span><span class="co">#&gt; 6                 0</span></span></code></pre></div>
<p>For an optimal execution of package, it is recommended to prepare
input file as below:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">input_segments</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MakeSegInput.html">MakeSegInput</a></span><span class="op">(</span>Segments     <span class="op">=</span> <span class="va">TCGA_BRCA_Segments</span>,</span>
<span>                               SampleID_col <span class="op">=</span> <span class="st">"SampleID"</span>, </span>
<span>                               Chr_col      <span class="op">=</span> <span class="st">"Chromosome"</span>, </span>
<span>                               Start_col    <span class="op">=</span> <span class="st">"Start"</span>, </span>
<span>                               End_col      <span class="op">=</span> <span class="st">"End"</span>, </span>
<span>                               MajorCN_col  <span class="op">=</span> <span class="st">"Major_Copy_Number"</span>, </span>
<span>                               MinorCN_col  <span class="op">=</span> <span class="st">"Minor_Copy_Number"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">input_segments</span><span class="op">)</span></span>
<span><span class="co">#&gt;           SampleID  Chr   Start     End MajorCN MinorCN CopyNumber  len_bp</span></span>
<span><span class="co">#&gt; 1 TCGA-D8-A27H-01A chr1   13116  732994       3       0          3  719878</span></span>
<span><span class="co">#&gt; 2 TCGA-D8-A27H-01A chr1  734210 1942014       2       2          4 1207804</span></span>
<span><span class="co">#&gt; 3 TCGA-D8-A27H-01A chr1 1943142 1987886       2       1          3   44744</span></span>
<span><span class="co">#&gt; 4 TCGA-D8-A27H-01A chr1 1989523 6821628       2       2          4 4832105</span></span>
<span><span class="co">#&gt; 5 TCGA-D8-A27H-01A chr1 6822926 6862432       1       1          2   39506</span></span>
<span><span class="co">#&gt; 6 TCGA-D8-A27H-01A chr1 6863666 6874235       2       0          2   10569</span></span>
<span><span class="co">#&gt;     len_mb</span></span>
<span><span class="co">#&gt; 1 0.719878</span></span>
<span><span class="co">#&gt; 2 1.207804</span></span>
<span><span class="co">#&gt; 3 0.044744</span></span>
<span><span class="co">#&gt; 4 4.832105</span></span>
<span><span class="co">#&gt; 5 0.039506</span></span>
<span><span class="co">#&gt; 6 0.010569</span></span></code></pre></div>
<p>In this function, you can easily create an input file based on your
data frame of allele-specific segments.</p>
<p><code>Segments</code> is the name of your segment variable which
should be a data frame.</p>
<p>Within your data frame, <code>SampleID_col</code> is the name of
corresponding column for the sample IDs, <code>Chr_col</code> is that
for chromosome name column, <code>Start_col</code> and
<code>End_col</code> are the start and end column of segments,
<code>MajorCN_col</code> and <code>MinorCN_col</code> are the name of
columns for major and minor copy numbers.</p>
</div>
<div class="section level2">
<h2 id="copy-number-signatures">Copy number signatures<a class="anchor" aria-label="anchor" href="#copy-number-signatures"></a>
</h2>
<p><code><a href="../reference/ExtractCnaSignature.html">ExtractCnaSignature()</a></code> function is used to classify
segments based on their copy numbers and length, and generate a list of
results contains the features for each constructed CNA signature and
contribution of each signature across the cohort.</p>
<p>To run this function, follow:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Should load NMF before this function</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://renozao.github.io/NMF/" class="external-link">NMF</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: package 'NMF' was built under R version 4.3.3</span></span>
<span><span class="co">#&gt; Loading required package: registry</span></span>
<span><span class="co">#&gt; Warning: package 'registry' was built under R version 4.3.3</span></span>
<span><span class="co">#&gt; Loading required package: rngtools</span></span>
<span><span class="co">#&gt; Warning: package 'rngtools' was built under R version 4.3.3</span></span>
<span><span class="co">#&gt; Loading required package: cluster</span></span>
<span><span class="co">#&gt; NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2</span></span>
<span><span class="co">#&gt;   To enable shared memory capabilities, try: install.extras('</span></span>
<span><span class="co">#&gt; NMF</span></span>
<span><span class="co">#&gt; ')</span></span>
<span><span class="va">result_signatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ExtractCnaSignature.html">ExtractCnaSignature</a></span><span class="op">(</span>Input.Segments <span class="op">=</span> <span class="va">input_segments</span>, </span>
<span>                                         Focal_th       <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                                         Broad_th       <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                                         Art.len        <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                         k_range        <span class="op">=</span> <span class="fl">2</span><span class="op">:</span><span class="fl">10</span>,</span>
<span>                                         n_run          <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; Compute NMF rank= 2  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 3  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 4  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 5  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 6  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 7  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 8  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 9  ... + measures ... OK</span></span>
<span><span class="co">#&gt; Compute NMF rank= 10  ... + measures ... OK</span></span>
<span><span class="co">#&gt; [1] "Estimating the best rank ..."</span></span></code></pre></div>
<p>The arguments needed for this function are
<code>Input.Segments</code> which is the data frame created by
<code><a href="../reference/MakeSegInput.html">MakeSegInput()</a></code>, <code>Focal_th</code> which is the
threshold of length in Mb for focal alterations, <code>Broad_th</code>
which is the threshold of length in Mb for broad alterations, and
<code>Art.len</code> which is the threshold for artifact segments which
are mainly due to hypersegmentation. Segments which are between focal
and broad thresholds are called medium segments. Other arguments are
<code>k_range</code> and <code>n_run</code> which are for NMF test.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Features of constructed signatures.</span></span>
<span><span class="va">signatures_features</span> <span class="op">&lt;-</span> <span class="va">result_signatures</span><span class="op">[[</span><span class="st">"Signature_Annotation"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">signatures_features</span><span class="op">)</span></span>
<span><span class="co">#&gt;             Signature Broad Hi-AMP Medium Hi-AMP Focal Hi-AMP Broad Mod-AMP</span></span>
<span><span class="co">#&gt; Signature1 Signature1 1.044879e-01  6.017817e-02 6.869141e-02  3.846139e-01</span></span>
<span><span class="co">#&gt; Signature2 Signature2 1.171759e-17  2.343519e-17 9.345572e-05  2.343519e-17</span></span>
<span><span class="co">#&gt; Signature3 Signature3 2.330925e-17  4.661849e-17 4.661849e-17  4.661849e-17</span></span>
<span><span class="co">#&gt; Signature4 Signature4 2.957284e-17  2.155282e-04 5.512496e-05  5.914567e-17</span></span>
<span><span class="co">#&gt; Signature5 Signature5 1.440680e-17  2.881361e-17 1.286929e-03  2.881361e-17</span></span>
<span><span class="co">#&gt; Signature6 Signature6 3.992851e-17  7.985702e-17 1.348897e-02  2.663088e-03</span></span>
<span><span class="co">#&gt;            Medium Mod-AMP Focal Mod-AMP   Broad Gain  Medium Gain  Focal Gain</span></span>
<span><span class="co">#&gt; Signature1   1.233033e-01  5.472604e-02 2.270972e-16 1.517705e-01 0.052228761</span></span>
<span><span class="co">#&gt; Signature2   2.343519e-17  2.916548e-04 2.343519e-17 2.343519e-17 0.001454432</span></span>
<span><span class="co">#&gt; Signature3   4.661849e-17  4.661849e-17 4.322520e-02 1.419514e-03 0.001487626</span></span>
<span><span class="co">#&gt; Signature4   7.961552e-03  6.666367e-03 8.485945e-01 8.532664e-02 0.025460235</span></span>
<span><span class="co">#&gt; Signature5   2.881361e-17  3.540745e-04 9.004653e-01 2.041908e-02 0.012488047</span></span>
<span><span class="co">#&gt; Signature6   6.758691e-03  1.880739e-02 1.197120e-04 4.223173e-01 0.214206343</span></span>
<span><span class="co">#&gt;               Broad DEL   Medium DEL    Focal DEL    Broad LOH   Medium LOH</span></span>
<span><span class="co">#&gt; Signature1 1.055295e-16 1.055295e-16 1.055295e-16 3.165885e-16 1.270832e-02</span></span>
<span><span class="co">#&gt; Signature2 1.171759e-17 1.171759e-17 1.073524e-04 3.515278e-17 9.441030e-14</span></span>
<span><span class="co">#&gt; Signature3 2.330925e-17 1.055508e-04 6.541292e-04 8.771279e-01 5.895267e-02</span></span>
<span><span class="co">#&gt; Signature4 2.957284e-17 2.957284e-17 2.957284e-17 3.761099e-01 5.159473e-02</span></span>
<span><span class="co">#&gt; Signature5 1.440680e-17 1.440680e-17 1.440680e-17 6.471974e-02 5.762722e-17</span></span>
<span><span class="co">#&gt; Signature6 3.788774e-03 5.014366e-03 8.125476e-03 2.687554e-03 2.570122e-01</span></span>
<span><span class="co">#&gt;               Focal LOH          NEU</span></span>
<span><span class="co">#&gt; Signature1 0.0026104574 1.055295e-16</span></span>
<span><span class="co">#&gt; Signature2 0.0034998264 9.945533e-01</span></span>
<span><span class="co">#&gt; Signature3 0.0199345807 2.330925e-17</span></span>
<span><span class="co">#&gt; Signature4 0.0191111194 2.957284e-17</span></span>
<span><span class="co">#&gt; Signature5 0.0002668796 1.440680e-17</span></span>
<span><span class="co">#&gt; Signature6 0.1102892751 3.992851e-17</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Contribution of each signature across the cohort</span></span>
<span><span class="va">signatures_contribution</span> <span class="op">&lt;-</span> <span class="va">result_signatures</span><span class="op">[[</span><span class="st">"Sample_Weights"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">signatures_contribution</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         Signature1  Signature2   Signature3   Signature4</span></span>
<span><span class="co">#&gt; 1105266               2.842008e-02 0.000406409 3.666116e-01 4.657297e-01</span></span>
<span><span class="co">#&gt; HCM-CSHL-0153-C50-01B 4.672040e-16 0.907483484 3.202799e-02 1.667199e-15</span></span>
<span><span class="co">#&gt; HCM-CSHL-0250-C50-01A 4.672040e-16 0.696625966 1.778519e-01 1.667199e-15</span></span>
<span><span class="co">#&gt; HCM-CSHL-0261-C50-01A 4.672040e-16 0.832339890 1.450671e-01 1.667199e-15</span></span>
<span><span class="co">#&gt; HCM-CSHL-0366-C50-01A 4.672040e-16 0.970521857 2.115204e-15 1.667199e-15</span></span>
<span><span class="co">#&gt; HCM-CSHL-0460-C50-01B 4.672040e-16 0.876581167 5.352163e-02 1.667199e-15</span></span>
<span><span class="co">#&gt;                       Signature5   Signature6</span></span>
<span><span class="co">#&gt; 1105266               0.07124511 6.757668e-02</span></span>
<span><span class="co">#&gt; HCM-CSHL-0153-C50-01B 0.04626705 1.422173e-02</span></span>
<span><span class="co">#&gt; HCM-CSHL-0250-C50-01A 0.12552282 1.234802e-15</span></span>
<span><span class="co">#&gt; HCM-CSHL-0261-C50-01A 0.01785011 4.743007e-03</span></span>
<span><span class="co">#&gt; HCM-CSHL-0366-C50-01A 0.02527071 4.207588e-03</span></span>
<span><span class="co">#&gt; HCM-CSHL-0460-C50-01B 0.03627716 3.362022e-02</span></span></code></pre></div>
<div class="section level3">
<h3 id="plot-signatures">Plot signatures<a class="anchor" aria-label="anchor" href="#plot-signatures"></a>
</h3>
<p>To visualise signatures’ features, you can pass
<code><a href="../reference/ExtractCnaSignature.html">ExtractCnaSignature()</a></code> results into
<code><a href="../reference/PlotSignatures.html">PlotSignatures()</a></code> as below:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signatures_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotSignatures.html">PlotSignatures</a></span><span class="op">(</span>Sig.Results <span class="op">=</span> <span class="va">result_signatures</span><span class="op">)</span></span></code></pre></div>
<p>Result is a list of bar plots for each signature:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">signatures_plots</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<p><img src="How-to-use_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="cytoband-level-alteration">Cytoband level alteration<a class="anchor" aria-label="anchor" href="#cytoband-level-alteration"></a>
</h2>
<p><code><a href="../reference/CytobandLevelAlteration.html">CytobandLevelAlteration()</a></code> function could accurately
detect cytoband level alteration using input file generated by
<code><a href="../reference/MakeSegInput.html">MakeSegInput()</a></code>. This function calculates the affected
length of each cytoband and estimates the length-weighted alterations
per cytoband in each sample. Currently, this function support both hg38
and hg19 reference genome.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cytoband_level</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CytobandLevelAlteration.html">CytobandLevelAlteration</a></span><span class="op">(</span>Input.Segments <span class="op">=</span> <span class="va">input_segments</span>,</span>
<span>                                          Genome         <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>                                          Art.len        <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cytoband_level</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 15</span></span></span>
<span><span class="co">#&gt;   SampleID band_chr band_start band_end band    band_span_mb covered_mb</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 1105266  chr1        2<span style="text-decoration: underline;">300</span>000  5<span style="text-decoration: underline;">400</span>000 1p36.32          3.1       3.10</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 1105266  chr1        5<span style="text-decoration: underline;">400</span>000  7<span style="text-decoration: underline;">200</span>000 1p36.31          1.8       1.80</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 1105266  chr1        7<span style="text-decoration: underline;">200</span>000  9<span style="text-decoration: underline;">200</span>000 1p36.23          2         2.00</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 1105266  chr1        9<span style="text-decoration: underline;">200</span>000 12<span style="text-decoration: underline;">700</span>000 1p36.22          3.5       3.50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 1105266  chr1       12<span style="text-decoration: underline;">700</span>000 16<span style="text-decoration: underline;">200</span>000 1p36.21          3.5       3.50</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 1105266  chr1       16<span style="text-decoration: underline;">200</span>000 20<span style="text-decoration: underline;">400</span>000 1p36.13          4.2       4.19</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 8 more variables: frac_band_covered &lt;dbl&gt;, tcn_lenw &lt;dbl&gt;, lcn_lenw &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   frac_del_lenw &lt;dbl&gt;, frac_loh_lenw &lt;dbl&gt;, frac_gain_lenw &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   frac_amp_mod_lenw &lt;dbl&gt;, frac_amp_hi_lenw &lt;dbl&gt;</span></span></span></code></pre></div>
<p>The arguments of function are <code>Input.Segments</code> which is
the data frame generated by <code><a href="../reference/MakeSegInput.html">MakeSegInput()</a></code>,
<code>Genome</code> which can be either <code>hg19</code> or
<code>hg38</code>, and <code>Art.len</code> which is the threshold for
artifact segments.</p>
<p>The output file has the below column:</p>
<table class="table">
<colgroup>
<col width="18%">
<col width="81%">
</colgroup>
<thead><tr class="header">
<th>Column</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>SampleID</td>
<td>Unique identifier of each sample</td>
</tr>
<tr class="even">
<td>band_chr</td>
<td>Chromosome name</td>
</tr>
<tr class="odd">
<td>band_start</td>
<td>Start point of cytoband</td>
</tr>
<tr class="even">
<td>band_end</td>
<td>End point of cytoband</td>
</tr>
<tr class="odd">
<td>band</td>
<td>Name of cytoband</td>
</tr>
<tr class="even">
<td>band_span_mb</td>
<td>Length of cytoband in Mb</td>
</tr>
<tr class="odd">
<td>covered_mb</td>
<td>Length of cytoband covered by corresponding segment in Mb</td>
</tr>
<tr class="even">
<td>frac_band_covered</td>
<td>Length of cytoband covered by corresponding segment in Mb</td>
</tr>
<tr class="odd">
<td>tcn_lenw</td>
<td>Length-weighted total copy number of cytoband</td>
</tr>
<tr class="even">
<td>lcn_lenw</td>
<td>Length-weighted minor copy number of cytoband</td>
</tr>
<tr class="odd">
<td>frac_del_lenw</td>
<td>Length-weighted fraction of cytoband which is deleted</td>
</tr>
<tr class="even">
<td>frac_loh_lenw</td>
<td>Length-weighted fraction of cytoband which harboures LOH</td>
</tr>
<tr class="odd">
<td>frac_gain_lenw</td>
<td>Length-weighted fraction of cytoband which has 2 &lt; copy number
&lt; 7</td>
</tr>
<tr class="even">
<td>frac_amp_mod_lenw</td>
<td>Length-weighted fraction of cytoband which has 6 &lt; copy number
&lt; 10</td>
</tr>
<tr class="odd">
<td>frac_amp_hi_lenw</td>
<td>Length-weighted fraction of cytoband which has copy number more than
or equal to 10</td>
</tr>
</tbody>
</table>
<div class="section level3">
<h3 id="genome-wide-alterations-plot">Genome wide alterations plot<a class="anchor" aria-label="anchor" href="#genome-wide-alterations-plot"></a>
</h3>
<p>In order to visualise genome wide alterations for Gain/Loss,
Amplification/Deletion, and rate of LOH across the cohort
<code><a href="../reference/PlotGenomeWideAlteration.html">PlotGenomeWideAlteration()</a></code> can be utilised using cytoband
level data frame created by <code><a href="../reference/CytobandLevelAlteration.html">CytobandLevelAlteration()</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genome_wide_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlotGenomeWideAlteration.html">PlotGenomeWideAlteration</a></span><span class="op">(</span>Cytoband.level.data <span class="op">=</span> <span class="va">cytoband_level</span>, </span>
<span>                                              Genome              <span class="op">=</span> <span class="st">"hg19"</span>, </span>
<span>                                              Coverage            <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<p>This function generates a list of Gain/Loss, Amp/Del, LOH rate plots,
and a data frame which contains the frequnecy of alterations per
cytoband.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genome_wide_plots</span><span class="op">[[</span><span class="st">"GainLoss_plot"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Warning: Removed 1 rows containing missing values (`geom_vline()`).</span></span></code></pre></div>
<p><img src="How-to-use_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genome_wide_plots</span><span class="op">[[</span><span class="st">"AmpDel_plot"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Warning: Removed 1 rows containing missing values (`geom_vline()`).</span></span></code></pre></div>
<p><img src="How-to-use_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genome_wide_plots</span><span class="op">[[</span><span class="st">"LOH_plot"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; Warning: Removed 1 rows containing missing values (`geom_vline()`).</span></span></code></pre></div>
<p><img src="How-to-use_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genome_wide_plots</span><span class="op">[[</span><span class="st">"Cytobands_frq"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 15</span></span></span>
<span><span class="co">#&gt;   chr      start    end band  Amp_pct Del_pct Gain_pct Loss_pct LOH_pct band_len</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;ord&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 1      2<span style="text-decoration: underline;">300</span>000 5.40<span style="color: #949494;">e</span>6 1p36…       0       0     53.7     46.3    56.1  3<span style="text-decoration: underline;">100</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 1      5<span style="text-decoration: underline;">400</span>000 7.20<span style="color: #949494;">e</span>6 1p36…       0       0     56.1     43.9    53.7  1<span style="text-decoration: underline;">800</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 1      7<span style="text-decoration: underline;">200</span>000 9.2 <span style="color: #949494;">e</span>6 1p36…       0       0     57.5     42.5    52.5  2<span style="text-decoration: underline;">000</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 1      9<span style="text-decoration: underline;">200</span>000 1.27<span style="color: #949494;">e</span>7 1p36…       0       0     59.0     41.0    51.3  3<span style="text-decoration: underline;">500</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 1     12<span style="text-decoration: underline;">700</span>000 1.62<span style="color: #949494;">e</span>7 1p36…       0       0     56.4     43.6    53.8  3<span style="text-decoration: underline;">500</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 1     16<span style="text-decoration: underline;">200</span>000 2.04<span style="color: #949494;">e</span>7 1p36…       0       0     53.8     46.2    56.4  4<span style="text-decoration: underline;">200</span>000</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5 more variables: cum_start_chr &lt;dbl&gt;, mid_local &lt;dbl&gt;, chr_len &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   chr_offset &lt;dbl&gt;, mid_cum &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by nima hemmat.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
