#' @title CytobandLevelAlteration()
#'
#' @description
#' Function to find cytoband level alteration in each sample using df generated by MakeSegInput()
#'
#' @param Input.Segments Segment data frame created by MakeSegInput()
#' @param Genome Whether samples are aligned to hg38 or hg19
#' @param Art.len Threshold for artifact segments (default = 0.05 Mb)
#'
#' @importFrom magrittr %>%
#'
#' @return A tibble containing cytoband level copy number alteration and affected fraction of each cytoband
#' @export

CytobandLevelAlteration <- function(Input.Segments, Genome = c("hg38","hg19"), Art.len = 0.05) {

  # Check input file and arguments
  if (missing(Input.Segments) || !is.data.frame(Input.Segments)) {
    stop("Input segment is not a data frame! Ideally, use MakeSegInput() output")
  }
  required_cols <- c("SampleID","Chr","Start","End","MajorCN",
                     "MinorCN","CopyNumber","len_bp","len_mb")
  if (!all(required_cols %in% colnames(Input.Segments))) {
    stop("Input data frame has not required columns")
  }
  acceptable_genome <- c("hg38", "hg19")
  if (!(Genome %in% acceptable_genome)) {
    stop("Only specify hg38 or hg19")
  }

  # Specify the version of cytoband list
  genome <- match.arg(Genome)
  cytoband_file <- GetCytoband(genome = genome)
  cytoband_file <- cytoband_file %>%
    dplyr::mutate(chr_new = gsub("chr", "", chr),
           band = paste0(chr_new, band)) %>%
    dplyr::select(.,-chr_new)

  # Unify chromosome prefix in both segments and cytoband files
  chr_unifier <- function(seg_df) {
    seg_df %>%
      dplyr::mutate(
        Chr = dplyr::if_else(grepl("^chr", Chr), Chr, paste0("chr", Chr))
      )
  }
  segs_unified <- chr_unifier(Input.Segments) %>%
    dplyr::filter(len_mb > Art.len)

  # Generate genomic range objects for both segments and cytobands
  segments_gr <- GenomicRanges::GRanges(
    seqnames = segs_unified$Chr,
    ranges   = IRanges::IRanges(start = segs_unified$Start,
                                end   = segs_unified$End)
    )
  S4Vectors::mcols(segments_gr)$SampleID <- segs_unified$SampleID
  S4Vectors::mcols(segments_gr)$MinorCN  <- segs_unified$MinorCN
  S4Vectors::mcols(segments_gr)$CopyNumber <- segs_unified$CopyNumber

  cytoband_gr <- GenomicRanges::GRanges(
    seqnames = cytoband_file$chr,
    ranges   = IRanges::IRanges(start = cytoband_file$start,
                                end   = cytoband_file$end)
  )

  S4Vectors::mcols(cytoband_gr)$band <- cytoband_file$band

  # Find overlaps between segments and cytobands
  hits <- IRanges::findOverlaps(cytoband_gr, segments_gr, type = "any", ignore.strand = TRUE)
  q <- S4Vectors::queryHits(hits)   # cytoband indices
  s <- S4Vectors::subjectHits(hits) # segment indices

  # intersected ranges and their length (in Mb)
  ovl <- IRanges::pintersect(IRanges::ranges(cytoband_gr)[q], IRanges::ranges(segments_gr)[s])
  len_mb <- IRanges::width(ovl) / 1e6

  # Gather overlaps
  overlaps <- tibble::tibble(
    SampleID      = S4Vectors::mcols(segments_gr)$SampleID[s],
    band_chr      = as.character(GenomicRanges::seqnames(cytoband_gr))[q],
    band_start    = IRanges::start(cytoband_gr)[q],
    band_end      = IRanges::end(cytoband_gr)[q],
    band          = S4Vectors::mcols(cytoband_gr)$band[q],
    seg_chr       = as.character(GenomicRanges::seqnames(segments_gr))[s],
    seg_start     = IRanges::start(segments_gr)[s],
    seg_end       = IRanges::end(segments_gr)[s],
    MinorCN       = S4Vectors::mcols(segments_gr)$MinorCN[s],
    CopyNumber    = S4Vectors::mcols(segments_gr)$CopyNumber[s],
    len_mb        = len_mb
  )

  # Band level for each sample
  band_level <- overlaps %>%
    dplyr::group_by(SampleID, band_chr, band_start, band_end, band) %>%
    dplyr::summarise(
      band_span_mb       = (dplyr::first(band_end) - dplyr::first(band_start)) / 1e6,
      covered_mb         = sum(len_mb),                               # total overlapped length
      frac_band_covered  = pmin(covered_mb / band_span_mb, 1),        # 0..1
      # length-weighted continuous summaries
      tcn_lenw           = sum(CopyNumber * len_mb) / sum(len_mb),
      lcn_lenw           = sum(MinorCN    * len_mb) / sum(len_mb),
      # length-weighted event fractions
      frac_del_lenw      = sum((CopyNumber == 0) * len_mb) / sum(len_mb),
      frac_loh_lenw      = sum((CopyNumber == 1 | (CopyNumber > 1 & MinorCN == 0)) * len_mb) / sum(len_mb),
      frac_gain_lenw     = sum((CopyNumber %in% c(3 , 4, 5, 6)) * len_mb) / sum(len_mb),
      frac_amp_mod_lenw  = sum((CopyNumber %in% c(7 , 8, 9)) * len_mb) / sum(len_mb),
      frac_amp_hi_lenw   = sum((CopyNumber > 9) * len_mb) / sum(len_mb),
      .groups = "drop"
  )
}
