#' @title PlotGenomeWideAlteration()
#'
#' @description
#' Function to graph genome-wide alteration based on cytoband level alteration table generated by CytobandLevelAlteration()
#'
#' @param Cytoband.level.data Data frame of cytoband alterations
#' @param Genome Whether samples are aligned to hg38 or hg19
#' @param Coverage Threshold for coverage of cytobands needed to be included in the graph (default = 0.8)
#'
#' @importFrom magrittr %>%
#'
#' @return A list of graphs showing the frequency of each alteration in each cytoband across the cohort
#' @export

PlotGenomeWideAlteration <- function(Cytoband.level.data,
                                     Genome = c("hg38","hg19"),
                                     Coverage = 0.8) {

  # Check input file and arguments
  if (missing(Cytoband.level.data) || !is.data.frame(Cytoband.level.data)) {
    stop("Input segment is not a data frame! Ideally, use CytobandLevelAlteration() output")
  }

  required_cols <- c("SampleID","band_chr","band_start","band_end","band",
                     "band_span_mb","covered_mb","frac_band_covered","tcn_lenw","lcn_lenw",
                     "frac_del_lenw","frac_loh_lenw","frac_gain_lenw","frac_amp_mod_lenw","frac_amp_hi_lenw")

  if (!all(required_cols %in% colnames(Cytoband.level.data))) {
    stop("Input data frame has not required columns")
  }

  acceptable_genome <- c("hg38", "hg19")
  if (!(Genome %in% acceptable_genome)) {
    stop("Only specify hg38 or hg19")
  }

  if (!is.numeric(Coverage) || Coverage > 1 || Coverage <= 0) {
    stop("Coverage of cytobands should be a number between 0 and 1")
  }

  # Specify the version of cytoband list
  genome <- match.arg(Genome)
  cytoband_file <- GetCytoband(genome = genome)

  # Modify cytoband file
  cytoband_file <- cytoband_file %>%
    dplyr::filter(chr %in% paste("chr", 1:22, sep = "")) %>%
    dplyr::mutate(chr = gsub("chr", "", chr),
           band = paste0(chr, band)) %>%
    dplyr::select(.,-gieStain)

  # Make a df containing the percentage of alteration of each cytoband across the cohort
  cytoband_lev_th <- Cytoband.level.data %>%
    dplyr::filter(frac_band_covered > Coverage) %>%
    dplyr::mutate(
      CN_status = dplyr::case_when(
        # Annotate
        frac_gain_lenw >= 0.8 ~"Gain",
        frac_amp_mod_lenw + frac_amp_hi_lenw >= 0.8 ~ "Amp",
        frac_del_lenw >= 0.8 ~ "Del",
        frac_loh_lenw >= 0.8 &
        frac_gain_lenw < 0.2 &
        frac_amp_mod_lenw < 0.2 &
        frac_amp_hi_lenw < 0.02 ~ "Loss",
        TRUE ~ "Mixed"
      ),
      LOH_status = dplyr::if_else(frac_loh_lenw >= 0.8, "LOH", "No LOH")
    ) %>%
    dplyr::filter(CN_status != "Mixed",
           !(band_chr %in% c("chrY", "chrX"))) %>%
    dplyr::group_by(band) %>%
    dplyr::summarise(
      Amp_pct = (sum(CN_status == "Amp") / dplyr::n()) * 100,
      Del_pct = (sum(CN_status == "Del") / dplyr::n()) * 100,
      Gain_pct = (sum(CN_status == "Gain")/ dplyr::n()) * 100,
      Loss_pct = 100 - (Amp_pct + Del_pct + Gain_pct),
      LOH_pct = (sum(LOH_status == "LOH") / dplyr::n() * 100),
      .groups = "drop"
    )

  # Make data required for graphing
  all_cytobands <- cytoband_file %>%
    dplyr::left_join(cytoband_lev_th, by = "band") %>%
    dplyr::mutate(dplyr::across(dplyr::where(is.numeric), ~dplyr::coalesce(.x, 0))) %>%
    dplyr::mutate(chr      = as.character(chr),
                  chr      = factor(chr, levels = as.character(1:22), ordered = TRUE),
                  band_len = end - start) %>%
    dplyr::arrange(chr, start) %>%
    dplyr::group_by(chr) %>%
    dplyr::mutate(
      cum_start_chr = cumsum(dplyr::lag(band_len, default = 0)),
      mid_local     = cum_start_chr + band_len / 2
    ) %>%
    dplyr::ungroup()

  # Calculate chromosome sizes
  chr_sizes <- all_cytobands %>%
    dplyr::group_by(chr) %>%
    dplyr::summarise(chr_len = sum(band_len), .groups = "drop") %>%
    dplyr::arrange(chr) %>%
    dplyr::mutate(chr_offset = cumsum(dplyr::lag(chr_len, default = 0)))

  all_cytobands <- all_cytobands %>%
    dplyr::left_join(chr_sizes, by = "chr") %>%
    dplyr::mutate(mid_cum = chr_offset + mid_local)

  # Calculate boundaries for chromosomes and labels on graphs
  chr_boundaries <- chr_sizes %>%
    dplyr::mutate(v_line = dplyr::if_else(chr != "22", chr_offset + chr_len, NA),
                  x_lab  = chr_offset + (chr_len / 2),
                  y_lab  = dplyr::if_else(as.numeric(chr) %% 2 == 0, -100, 100))

  # Gain and Losses
  GainLoss_plot <- ggplot2::ggplot(all_cytobands, ggplot2::aes(x = mid_cum)) +
    ggplot2::geom_density(ggplot2::aes(y = Gain_pct, fill = "Gain"), stat = "identity", color = NA) +
    ggplot2::geom_density(ggplot2::aes(y = -Loss_pct, fill = "Loss"), stat = "identity", color = NA) +
    ggplot2::scale_fill_manual(values = c("Gain" = "red",
                                          "Loss" = "blue")) +
    ggplot2::scale_y_continuous(limits = c(-100, 100),
                                breaks = c(-100, -75, -50, -25, 0, 25, 50, 75, 100),
                                labels = function(x) abs(x)) +
    ggplot2::scale_x_continuous(expand = c(0.002, 0.002)) +
    ggplot2::geom_vline(data = chr_boundaries,
                        ggplot2::aes(xintercept = v_line),
                        lwd = 0.2,
                        linetype = "longdash",
                        color = "gray") +
    ggplot2::geom_hline(yintercept = 0, color = "gray30", lwd = 0.2) +
    ggplot2::geom_text(data = chr_boundaries,
                       ggplot2::aes(x = x_lab, label = chr, y = y_lab),
                       size = 2.5) +
    ggplot2::labs(y = "Samples(%)", x = "Chromosomes") +
    ggplot2::theme_bw() +
    ggplot2::theme(
      panel.grid      = ggplot2::element_blank(),
      axis.text.x     = ggplot2::element_blank(),
      axis.ticks.x    = ggplot2::element_blank(),
      legend.title    = ggplot2::element_blank(),
      legend.position = "bottom"
    )

  # Amplifications and Deletions
  AmpDel_plot <- ggplot2::ggplot(all_cytobands, ggplot2::aes(x = mid_cum)) +
    ggplot2::geom_density(ggplot2::aes(y = Amp_pct, fill = "Amplification"), stat = "identity", color = NA) +
    ggplot2::geom_density(ggplot2::aes(y = -Del_pct, fill = "Deletion"), stat = "identity", color = NA) +
    ggplot2::scale_fill_manual(values = c("Amplification" = "red",
                                          "Deletion"      = "blue")) +
    ggplot2::scale_y_continuous(limits = c(-100, 100),
                                breaks = c(-100, -75, -50, -25, 0, 25, 50, 75, 100),
                                labels = function(x) abs(x)) +
    ggplot2::scale_x_continuous(expand = c(0.002, 0.002)) +
    ggplot2::geom_vline(data = chr_boundaries,
                        ggplot2::aes(xintercept = v_line),
                        lwd = 0.2,
                        linetype = "longdash",
                        color = "gray") +
    ggplot2::geom_hline(yintercept = 0, color = "gray30", lwd = 0.2) +
    ggplot2::geom_text(data = chr_boundaries,
                       ggplot2::aes(x = x_lab, label = chr, y = y_lab),
                       size = 2.5) +
    ggplot2::labs(y = "Samples(%)", x = "Chromosomes") +
    ggplot2::theme_bw() +
    ggplot2::theme(
      panel.grid      = ggplot2::element_blank(),
      axis.text.x     = ggplot2::element_blank(),
      axis.ticks.x    = ggplot2::element_blank(),
      legend.title    = ggplot2::element_blank(),
      legend.position = "bottom"
    )

  chr_boundaries_loh <- chr_sizes %>%
    dplyr::mutate(v_line = dplyr::if_else(chr != "22", chr_offset + chr_len, NA),
                  x_lab  = chr_offset + (chr_len / 2),
                  y_lab  = dplyr::if_else(as.numeric(chr) %% 2 == 0, 98, 100))
  # LOH
  LOH_plot <- ggplot2::ggplot(all_cytobands, ggplot2::aes(x = mid_cum)) +
    ggplot2::geom_density(ggplot2::aes(y = LOH_pct, fill = "Loss of heterozygosity"), stat = "identity", color = NA) +
    ggplot2::scale_fill_manual(values = c("Loss of heterozygosity" = "seagreen2")) +
    ggplot2::scale_y_continuous(limits = c(0, 100),
                                breaks = c(0, 25, 50, 75, 100)) +
    ggplot2::scale_x_continuous(expand = c(0.002, 0.002)) +
    ggplot2::geom_vline(data = chr_boundaries_loh,
                        ggplot2::aes(xintercept = v_line),
                        lwd = 0.2,
                        linetype = "longdash",
                        color = "gray") +
    ggplot2::geom_hline(yintercept = 0, color = "gray30", lwd = 0.2) +
    ggplot2::geom_text(data = chr_boundaries_loh,
                       ggplot2::aes(x = x_lab, label = chr, y = y_lab),
                       size = 2.5) +
    ggplot2::labs(y = "Samples(%)", x = "Chromosomes") +
    ggplot2::theme_bw() +
    ggplot2::theme(
      panel.grid      = ggplot2::element_blank(),
      axis.text.x     = ggplot2::element_blank(),
      axis.ticks.x    = ggplot2::element_blank(),
      legend.title    = ggplot2::element_blank(),
      legend.position = "bottom"
    )

  # Gather all plots in a list
  AlterationsPlots <- list(
    GainLoss_plot = GainLoss_plot,
    AmpDel_plot   = AmpDel_plot,
    LOH_plot      = LOH_plot,
    Cytobands_frq = all_cytobands
  )

  return(AlterationsPlots)

}
